mbed_tools_ci/utils/filesystem_helpers.py

```diff
@@ -77,3 +81,56 @@ def find_file_in_tree(file_name: str,
             f'File [{file_name}] not found anywhere in directories {"above" if top else "under"} {starting_point}'
         )
     return file_path
+
+
+class TemporaryDirectory:
+    """Creates and returns a temporary directory.
+
+    This is the same as tempfile.TemporaryDirectory in most cases.
+    However, on CI the temporary folder may not be accessible.
+    Moreover, on Windows TemporaryDirectory will fail cleaning up
+    (https://bugs.python.org/issue22107) and hence, a workaround had to be
+    implemented.
+
+    """
+
+    def __init__(self, path: Optional[str] = None) -> None:
+        """Constructor.
+
+        Args:
+            path: path to the temporary directory. If not set,
+            the platform temporary folder will be used if accessible.
+        """
+        self._tmp_dir_context_manager = None if path else tempfile.TemporaryDirectory()
+        _path = self._tmp_dir_context_manager.name if self._tmp_dir_context_manager else path
+        if not _path:
+            raise ValueError("Undefined temporary directory.")
+        self._tmp_dir_path = Path(_path).resolve()
+
+        if not self._tmp_dir_path.exists():
+            self._tmp_dir_path.mkdir(exist_ok=True)
+
+    @property
+    def path(self) -> Path:
+        """Path to the folder."""
+        return self._tmp_dir_path
+
+    def __str__(self) -> str:
+        """String representation."""
+        return str(self._tmp_dir_path)
+
+    def __enter__(self) -> Path:
+        """Context manager entry point."""
+        return self.path
+
+    def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
+        """Context manager exit point."""
+        self.cleanup()

```

Comment:
Why not put `cleanup` body in here?

Reply:
just did it

Reply:
Had to roll back this change as cleanup method is called by the GitTempClone class and this method is clearer to call thant .__exit__(.....)